### PR TITLE
Add logging of gRPC failures in case where problem is from server-sid…

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Adds logging of those Bridge call failures that are NOT due to client making invalid call but that appear server (Bridge) side failure.

**Which issue(s) this PR fixes**:
-

**Checklist**
- [x] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by ivansenic** [comment]
I added a fix for CI in https://github.com/stargate/stargate/pull/2171/commits/b2e407f62780ca56648a07e4b069bd27e7bcbab3, that was introduced by #2168..

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java [hunk 1]
```diff
@@ -24,9 +24,12 @@
...
 import org.jboss.resteasy.reactive.server.ServerExceptionMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 /** Simple exception mapper for the {@link StatusRuntimeException}. */
 public class StatusRuntimeExceptionMapper {
+  private static final Logger LOGGER = LoggerFactory.getLogger(StatusRuntimeExceptionMapper.class);
 
   @ServerExceptionMapper
   public RestResponse<ApiError> statusRuntimeException(StatusRuntimeException grpcE) {
```
Comment (ivansenic) [inline apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java:35]: todo below can be removed

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java [hunk 2]
```diff
@@ -39,35 +42,35 @@ public RestResponse<ApiError> statusRuntimeException(StatusRuntimeException grpc
...
-        return grpcResponse(
+        return grpcResponseNoLog(
             sc, Response.Status.UNAUTHORIZED, "Unauthorized gRPC operation", grpcE.getMessage());
       case UNAUTHENTICATED:
-        return grpcResponse(
+        return grpcResponseNoLog(
             sc, Response.Status.UNAUTHORIZED, "Unauthenticated gRPC operation", grpcE.getMessage());
-      case UNAVAILABLE:
+      case UNAVAILABLE: // server-side problem, do log
         // Should this be SERVICE_UNAVAILABLE (503) or BAD_GATEWAY (502)?
```
Comment (ivansenic) [inline apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java:66]: We should be correct wrt to HTTP status code and actually return bad gateway here

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java [hunk 3]
```diff
@@ -88,15 +91,34 @@ public RestResponse<ApiError> statusRuntimeException(StatusRuntimeException grpc
...
+        grpcCode, httpStatus, buildFailureMessage(grpcCode, httpStatus, prefix, suffix));
+  }
+
+  private static String buildFailureMessage(
+      Status.Code grpcCode, Response.Status httpStatus, String prefix, String suffix) {
+    return String.format(
+        "%s (Status.Code.%s->%d): %s", prefix, grpcCode, httpStatus.getStatusCode(), suffix);
+  }
+
+  private static RestResponse<ApiError> buildGrpcResponse(
```
Comment (ivansenic) [inline apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java:118]: should be `buildRestResponse`?

**Review by ivansenic — APPROVED** [review]

**Comment by ivansenic** [comment]
@tatu-at-datastax I will push small code fixes and merge this so I can continue with the release..

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java (hunk 1 above)
Comment (ivansenic) [reply apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java]: Done in https://github.com/stargate/stargate/pull/2171/commits/eb812454b4c804ffb27004f518b852ea9e01ed5b

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java (hunk 2 above)
Comment (ivansenic) [reply apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java]: Done in https://github.com/stargate/stargate/pull/2171/commits/eb812454b4c804ffb27004f518b852ea9e01ed5b

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java (hunk 3 above)
Comment (ivansenic) [reply apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java]: Done in https://github.com/stargate/stargate/pull/2171/commits/eb812454b4c804ffb27004f518b852ea9e01ed5b

File: apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ivansenic) [reply apis/sgv2-quarkus-common/src/main/java/io/stargate/sgv2/api/common/exception/StatusRuntimeExceptionMapper.java]: Done in https://github.com/stargate/stargate/pull/2171/commits/eb812454b4c804ffb27004f518b852ea9e01ed5b

**Comment by tatu-at-datastax** [comment]
Thank you @ivansenic I should have added a note that this is fine to change & merge since I am out and we need the release. Good points on other places that similarly need logging.

**Pull request merged**